docs/_tutorials/spring-cloud-autoconf-jms.md

```diff
@@ -0,0 +1,341 @@
+
+![overview]({{ site.baseurl }}/images/spring-cloud-app-architecture.png){: .center-image}
+
+## Goals
+
+The goal of this tutorial is to demonstrate auto injecting a [JMS ConnectionFactory](https://docs.solace.com/API-Developer-Online-Ref-Documentation/jms/javax/jms/ConnectionFactory.html based on  the application's Cloud Foundry Service Bindings and connect to the Solace Messaging service instance.  This tutorial will show you:
+
+1. How to Autowire a [JMS ConnectionFactory](https://docs.solace.com/API-Developer-Online-Ref-Documentation/jms/javax/jms/ConnectionFactory.html) into your application
+1. How to Autowire the [SolaceMessagingInfo](https://github.com/SolaceProducts/sl-spring-cloud-connectors/blob/master/src/main/java/com/solace/spring/cloud/core/SolaceMessagingInfo.java ) provided by the Cloud Foundry environment using Spring Cloud Connectors.
+1. How to Autowire [SpringSolJmsConnectionFactoryCloudFactory](https://github.com/SolaceProducts/solace-jms-spring-boot/blob/master/solace-jms-spring-boot-autoconfigure/src/main/java/com/solacesystems/jms/SpringSolJmsConnectionFactoryCloudFactory.java ) which you can use to access other Cloud Available Solace Messaging Instances and create other Solace implementations of the JMS ConnectionFactory.
+1. How to establish a connection to the Solace Messaging service.
+1. How to publish, subscribe and receive messages.
+
+## Assumptions
+
+This tutorial assumes the following:
+
+* You are familiar with Solace [core concepts]({{ site.docs-core-concepts }}){:target="_top"}.
+* You are familiar with [Spring RESTful Web Services](https://spring.io/guides/gs/rest-service/){:target="_blank"}.
+* You are familiar with [Cloud Foundry](https://www.cloudfoundry.org/){:target="_blank"}.
+* You have access to a running Pivotal Cloud Foundry environment.
+* Solace Messaging for PCF has been installed in your Pivotal Cloud Foundry environment.
+
+## Obtaining the Solace API
+
+This tutorial depends on you having the Solace Messaging API for JMS. Here are a few easy ways to get the JMS API. The instructions in the [Building](#building) section assume you're using Gradle and pulling the jars from maven central. If your environment differs then adjust the build instructions appropriately.
+
+### Get the API: Using Gradle
+
+```
+compile("com.solacesystems:sol-jms:${solaceJMSVersion}")
+```
+
+### Get the API: Using Maven
+
+```
+<dependency>
+  <groupId>com.solacesystems</groupId>
+  <artifactId>sol-jms</artifactId>
+  <version>{{ site.jms_version }}</version>
+</dependency>
+```
+
+### Get the API: Using the Solace Developer Portal
+
+The JMS API library can be [downloaded here]({{ site.links-downloads }}){:target="_top"}. The JMS API is distributed as a zip file containing the required jars, API documentation, and examples. 
+
+## Code Walk Through
```

Comment:
The "Code Walk Through" title is not a title we use in our samples. This isn't meant to be a code walk through. It's meant to be a tutorial which teaches a concept using code.